nrf5340 qspi xip sample migration to DTS partitining (#29680)
nrfconnect/sdk-nrf: doc/nrf/app_dev/device_guides/nrf53/qspi_xip_guide_nrf5340.rst

    Due to QSPI peripheral product anomaly, the QSPI peripheral must be run with the ``HFCLK192MCTRL=0`` setting.
    Any other value may cause undefined operation of the device.

Setting up NVM memory layout with devicetree
********************************************

> You need to complete the setup to use the expected NVM memory layout with devicetree partitioning.

Reviewer: ```suggestion
Complete a setup to use the expected NVM memory layout with devicetree partitioning.
```

The configuration must have 3 images with 2 slots each:

> .. figure:: images/nrf5340_static_partition_manager_slots.svg

Reviewer 2: This image needs to be updated - there is no such thing like mcuboot_primary with the DTS-based partitioning.

   :alt: Static partitioning slots in the nRF5340 SoC

   Partitioning slots in the nRF5340 SoC.

* The first set of slots is for the internal flash part of the application.
  These slots should be nodes with node labels ``slot0_partition`` and ``slot1_partition``.
* The second set of slots is for the network core update.

>   These slots should be nodes with node labels ``slot2_partition`` and ``slot3_partition``.

Author: `nodes with node labels` sound overcomplicated, maybe i can name it as `partitions named` or `partition node`
@annwoj have you any suggestion?

  The partition node ``slot2_partition`` is defined in the MCUboot flash simulator devicetree overlay.
  You only need to align its size with ``slot3_partition`` to match the expected NVM memory layout.
  Both slots should be the same size as the ``s0_partition`` partition node in the ``cpunet`` images.
* The third set of slots is for the QSPI XIP part of the application.

>   Use the partition nodes ``slot4_partition`` and ``slot5_partition``.

Reviewer 2: Should they be included in one of the figures too?

See the following snippet for an example of the memory layout configuration for devicetree:

.. literalinclude:: ../../../../../samples/nrf5340/extxip_smp_svr/boards/nrf5340dk_nrf5340_cpuapp.overlay
    :language: dts

> Partitioning for the ``cpunet`` images (``b0n`` and ``hci_ipc``) is expected to follow the normal nRF5340 SoC configuration.

Reviewer: ```suggestion
Partitioning for the ``cpunet`` images (``b0n`` and ``hci_ipc``) is expected to follow the nRF5340 SoC configuration.
```
